### DISCUSSION TITLE
how to get some information that is internal to HomeSpan inside my code?

### DISCUSSION THREAD

**Discussion by homonto** [HomeSpan/HomeSpan#722] [Q&A] [answered]
Hi guys,
I am not a guru in C++ so I am wondering if you could help me.
There are some methods (is this a good name?) used by the library but I am not sure how to get to this information i my code.
Examples:
- on weblog there is an info about wifi disconnect (WiFi Disconnects:). I found it in the library as:
`String(homeSpan.connected/2)`
but homeSpan.connected is not accessible in the code out of the library
- there are few others but I am quite sure if I know how to get this one, I would manage the others as well

Kindly asking for help - thx. ;-)

**Comment by HomeSpan** [comment]
In C++ variables can be public or private.  Public variables are accessible by those using the class and are guaranteed to have the same meaning all the time, they are documented, etc.  Private variables are for internal use only, and the developer can change them at any time, or even delete them.  More typical is that *all* variables in a class will be private, and users of the class will only interact with a small subset of those variables through methods like getVariable() and setVariable().  You'll note that HomeSpan uses this extensively.

If there is a get/set function, it means I'm providing access to a variable that will be guaranteed to be consistent, documented, etc.  All others are subject to change/disappear at any time with no notice as part of the development process.  For example, I've changed how I use the internal variable connected many times, including the fact that it once was a boolean and now it is an integer.

**Comment by homonto** [comment]
OK, I think I got it
If there is no `public:`  then it is private by default, right?
it means: the class Span everything until "public:" keyword is private
connected - private
but updateDatabase(..) - public
setApPassword - public - and that is how I can use it: 
`homeSpan.setApPassword("");   // no password for CP`
right?

**Comment by HomeSpan** [comment, marked as answer]
That's right.  For *classes*, the default is everything is *private* unless you precede with *public* or *protected* (you can also explicitly say *private* to make sure it's clear).  The *protected* state means that a variable cannot be access externally, but it can be accessed by any new class that is derived from the base class.

For *structures* (which are nearly the same as *classes*) everything is always *public* by default, unless you add the words *private* or *protected*.

**Reply by homonto** [reply]
Thank you very much. I know these are not HomeSpan questions but still - thank you for your help.
I was actually looking for "private" keyword but I was not aware "not public = private" ;-)
